tests/foreman/api/test_subnet.py

```diff
@@ -0,0 +1,374 @@
+"""Tests for the ``subnets`` paths.
+
+An API reference is available here:
+http://theforeman.org/api/apidoc/v2/1.15.html
+
+
+:Requirement: Subnet
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from robottelo.decorators import stubbed, tier1, tier2, tier3
+from robottelo.test import APITestCase
+from robozilla.decorators import skip_if_bug_open
+
+
+class ParameterizedSubnetTestCase(APITestCase):
+    """Implements parametrized subnet tests in API"""
+
+    @stubbed
+    @tier1
+    def test_positive_add_parameter(self):
+        """Parameters can be created in subnet
+
+        :id: c1dae6f4-45b1-45db-8529-d7918e41a99b
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter with single key and single value
+
+        :expectedresults: The parameter should be created in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_add_parameter_with_multiple_values(self):
+        """Subnet parameters can be created with multiple values
+
+        :id: b3de6f96-7c39-4c44-b91c-a6d141f5dd6a
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having single key and multiple values
+                separated with comma
+
+        :expectedresults: The parameter with multiple values should be saved
+            in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_parameter_and_valid_separator(self):
+        """Subnet parameters can be created with multiple names with valid
+        separators
+
+        :id: d1e2d75a-a1e8-4767-93f1-0bb1b75e10a0
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having key with multiple names separated
+                by valid separators(e.g fwd slash) and value
+
+        :expectedresults: The parameter with multiple names separated by valid
+            separators should be saved in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_create_with_parameter_and_invalid_separator(self):
+        """Subnet parameters can not be created with multiple names with
+        invalid separators
+
+        :id: 08d10b75-a0db-4a11-a915-965a2a207d16
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having key with multiple names separated
+                by invalid separators(e.g spaces) and value
+
+        :expectedresults: The parameter with multiple names separated by
+            invalid separators should not be saved in subnet
```

Comment:
and valid error should be raised.. In all -ve tests we would like to see a valid error message.